forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -4496,8 +4500,9 @@ public final void executePerpetual(Map<String, Object> p) {
                 false, (Set<RemoveType>) p.get("RemoveXTypes"),
                 (long) p.get("Timestamp"), (long) 0, true, false);
         } else if (category.equals("Colors")) {
-            addColor((ColorSet) p.get("Colors"), !(boolean) p.get("Overwrite"), (long) p.get("Timestamp"),
-                (long) 0, false);
+            ColorSet colors = (ColorSet) p.get("Colors");
+            addColor(colors, !(boolean) p.get("Overwrite"), (long) p.get("Timestamp"), (long) 0, false);
+            getCurrentState().addColor(colors.getColor());
```

Comment:
why do we need to hard set the states now?
that seems wrong

Reply:
I was encountering an issue whereupon being cast, cards would not display as their updated colors anymore.

So after the Animate, everything looked fine while the card was in the hand. But then upon cast, there was a misalignment between the colors displayed in Detail Pane / "Current Card Colors" (also the color coding of the spell on the stack) and the actual card item when checked in the debugger. So in the debugger the card object did have the new color identity... it just wasn't displaying properly in the GUI.

Is there a less aggressive fix?

Reply:
if it's a GUI problem only then the logic where the view gets refreshed probably needs updating
chances are its just using the state directly because it really doesn't expect what you're doing there - since the implementation you're "abusing" was really only meant for _Volrath's Shapeshifter_

Reply:
I think I've dodged this now